### PR TITLE
Fix bug with partial Hiredis availability

### LINKED ISSUE #3392: Seeking guidance for `hiredis` dependency on `redis-py`
I have a question regarding the `hiredis` dependency in `redis-py`. 
Is hiredis a required dependency for using the package, or is it optional? 

I have an implicit dependency on `redis-py`  and I've encountered an issue with a locked version of `hiredis` in my `requirements.txt`

Since `redis-py` updates automatically as it is implicit, it can lead to a version mismatch where `redis-py` is updated, but `hiredis` remains locked to an older version, causing an unsynchronized state.

Could you provide guidance on how to handle this situation and avoid any conflicts arising from the version lock?

### PR DESCRIPTION
**Pull request by vladvildanov**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Do tests and lints pass with this change?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

Closes #3392

### CODE REVIEW

File: redis/utils.py [hunk 1]
```diff
@@ -8,7 +8,13 @@
 
     # Only support Hiredis >= 3.0:
     HIREDIS_AVAILABLE = int(hiredis.__version__.split(".")[0]) >= 3
+    if not HIREDIS_AVAILABLE:
+        raise ImportError("hiredis package should be >= 3.0.0")
     HIREDIS_PACK_AVAILABLE = hasattr(hiredis, "pack_command")
```
Comment (dmaier-redislabs) [inline redis/utils.py:13]: Can it actually happen in `hiredis-py` doesn't have the `pack` command with versions >=3?

**Comment by dmaier-redislabs** [comment]
Hi Vlad, it seems that the Code linter test is failing because of an unused import.

File: redis/utils.py (hunk 1 above)
Comment (vladvildanov) [reply redis/utils.py]: Check was removed

**Review by dmaier-redislabs — APPROVED** [review]
Looks good to me

**Pull request merged**